Added support for accented characters

FrivalszkyP opened the pull request:
I added an iconv conversion to every matched string so that accented characters are transliterated to their unaccented versions before searching. This is simple multilanguage support and this is how Google works, for example. For example if you search for "perez" it will return results containing "pérez" too.

I don't think it is necessary for this to be a configurable option because I can't really think of a situation where this would not come in handy.

mahagr commented:
Maybe there's a need to whitelist languages where this is allowed. This only works with languages that use latin characters.

FrivalszkyP commented:
Right, right. It should be tested with non-latin alphabets.

FrivalszkyP commented:
I added a config option so that you can add this to your simplesearch.yaml file: 
```
ignore_accented_characters: true
```
This way we don't have to maintain a whitelist of allowed languages etc. but the site administrator can decide if they want this feature or not. 

I also refactored the code a little bit so it is more maintainable but it also comes with a little bit of overhead. I think it's acceptable.

flaviocopes commented:
Looks good, the only thing is, add an entry for the option in the blueprints.yaml file, so it's configurable from the Admin interface

FrivalszkyP commented:
Added the changes, if you like it, please do a squash & merge if possible!

flaviocopes commented:
Merged, thanks!

A---- commented:
The `en_US` locale has to be installed for this feature to work, or it wont be able to properly transliterate characters: `string(5) "p?rez"`

(Also, this option could be documentated in the `README.md`.)

flaviocopes commented:
Documented
